tests/test_api.py

```diff
@@ -194,6 +198,180 @@ def test_action_output(mock_boto, mock_get_snapshot_ids):
                 keep=["Image-20250104", "Image-20250102"],
                 skip=["Image-20250103"],
             ),
-            policy={"action": "deprecate", "keep": 2},
+            policy={"action": "deprecate", "keep": 2, "expiration_months": None},
+        )
+    }
+
+
+@patch("ami_deprecation_tool.api._get_snapshot_ids")
+@patch("ami_deprecation_tool.api.boto3")
+def test_depreciate_images_past_expiration(mock_boto, mock_get_snapshot_ids):
```

Comment:
the last thing worth considering is, afaict, the logical items are the same with the input data being different. there are a couple things you could do here (i wouldn't do both):

1. refactor the various assignments and assertions into a function. feed test specific data into that function
2. use `pytest.mark.parametrize` with different inputs. if you go with this approach, i would suggest defining a "EXPIRATION_POLICY_SCENARIOS" dictionary to make it clear what the cases are and to make it easy to identify grouped objects